Move case-search addons to USH FF

sravfeyn opened the pull request:
## Summary
https://dimagi-dev.atlassian.net/browse/USH-732

## Feature Flag

USH Case Search addons

## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan
https://dimagi-dev.atlassian.net/browse/QA-2882

### Safety story
<!--
Describe any other pieces to the safety story including
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.
-->

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

sravfeyn committed:
Move case-search addon features to USH ff

sravfeyn committed:
Move USH specific addon to FF

sravfeyn commented:
@orangejenny

orangejenny commented on corehq/toggles.py:
```diff
@@ -2026,3 +2026,12 @@ def _commtrackify(domain_name, toggle_is_enabled):
 )
+
+
+USH_CASE_CLAIM_ADDONS = StaticToggle(
```
I thought we were going to rename the existing `CASE_CLAIM_AUTOLAUNCH` flag, rather than have two USH-specific feature flags.

sravfeyn replied:
We discussed this a bit on the ticket https://dimagi-dev.atlassian.net/browse/USH-732

My understanding was that there are other projects that should be able to use regular case-search via `CASE_CLAIM_AUTOLAUNCH` flag and additional case-search config functionality on top of the features provided by `CASE_CLAIM_AUTOLAUNCH` will be under the new  `USH_CASE_CLAIM_ADDONS` flag. You can see the discussion under the ticket. Do let me know if my understanding is incorrect.

orangejenny replied:
Regular case claim is under `SYNC_SEARCH_CASE_CLAIM`, only USH projects are using `CASE_CLAIM_AUTOLAUNCH`

sravfeyn replied:
Cool. I have updated now.

sravfeyn committed:
Use auto-lanch toggle as USH specific toggle

sravfeyn committed:
rename toggle var

sravfeyn committed:
Merge master

sravfeyn committed:
truer description

sravfeyn committed:
disable configured features on toggle at build time

sravfeyn committed:
pass default values

sravfeyn committed:
correct

sravfeyn committed:
fix tests

sravfeyn commented:
QA Passed now.

Charl1996 committed:
Pull master in branch

orangejenny committed:
Renamed old flag in one more place

orangejenny committed:
Fixed bad merge

This and the previous commit are to fix a bad merge from 8955b622dfe1cd3b4f7f9be5e50aab225f393d51
